### ISSUE TITLE
[HOLD for payment 2024-05-08] HIGH: [Polish] [$500] Broken message text in LHN when Inviting someone to room. 

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#32978]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.12.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @marcaaron 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1702330874424759

## Action Performed:
1. Create a new workspace
2. Create a workspace room
3. Invite a workspace member to the room
4. Invite a non-workspace member to the room
5. View the chat as the non-workspace member

## Expected Result:
The last message text correctly shows "invited 2 users"

## Actual Result:
the message text says "invited  and"

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

![Snip - (5) New Expensify - Google Chrome](https://github.com/Expensify/App/assets/38435837/f7cd8b74-3d18-4a2e-ae04-c7e86e4ed70c)

![invited chat text weird](https://github.com/Expensify/App/assets/38435837/34a771f1-63a1-4c07-8a47-01a525e0a334)

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d4986375431dd52a</li>
        <li>Upwork Job ID: 1734953294535315456</li>
        <li>Last Price Increase: 2024-02-06</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Contributor | 0</li>
<li>dukenv0307 | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sakluger</details>

**Comment by artus9033** [comment]
Seems like this issue is non-reproducible anymore on the current `main` branch:
<img width="383" alt="image" src="https://github.com/Expensify/App/assets/10586109/a2ad47bc-e1c8-4e4d-a1d8-59da466ace1b">
<img width="379" alt="image" src="https://github.com/Expensify/App/assets/10586109/9e70cf80-c315-401e-aa50-7425f6eb1b77">

**Comment by sakluger** [comment]
Thanks @artus9033.

@m-natarajan are you still able to reproduce this issue?

**Comment by sakluger** [comment]
I also can't reproduce but may be following the steps incorrectly. I asked in the original Slack thread for help checking on reproduction.

**Comment by sakluger** [comment]
Closing as non-reproduceable.

**Comment by kbecciv** [comment]
@sakluger QA team found the reproducible steps for this issue:
1. Create a workspace room.
2. Invite users.
3. Send a message to another chat (if on web).
4. Log out and log in.
5. Refresh the page.
Can you try again, please? Thank you!
https://github.com/Expensify/App/assets/93399543/35f4287a-7e68-4631-9e8b-f9fe5948d36d

**Comment by sakluger** [comment]
@artus9033 are you able to reproduce with the above steps? If so, would you like to submit a proposal to fix?

**Comment by techievivek** [comment]
@aimane-chnaif Quick question: Does the bug above gets fixed if we refresh the page or does it persist?

**Comment by techievivek** [comment]
Ok, great, thanks for confirming.

**Comment by techievivek** [comment]
I am able to reproduce this. Looking into the backend logic.
<img width="359" alt="Screenshot 2024-01-15 at 3 33 37 PM" src="https://github.com/Expensify/App/assets/35863227/35d3f640-74e4-45c3-8dec-148d92556eed">

**Comment by techievivek** [comment]
We construct the mention-user message correctly, but we strip out the HTML-tag-like structure when we convert the HTML to text. As a result, the content is reduced to commas and space.


<img width="938" alt="Screenshot 2024-01-15 at 3 37 22 PM" src="https://github.com/Expensify/App/assets/35863227/932c4422-762d-4337-8575-e5909700197e">

@jasperhuangg Looks like you worked on the implementation: https://github.com/Expensify/Web-Expensify/pull/39506/files, Do you have any thoughts on handling this better?

**Comment by techievivek** [comment]
I am thinking maybe frontend can handle this differently i.e. rather than using the `text` content for LHN, they can parse the HTML content following this https://github.com/Expensify/App/blob/2be282e9ab2fa9aa289f298215b895e21dd4e01b/src/pages/home/report/ReportActionItemMessage.tsx#L39-L40

Because this seems broken only for LHN, the report action in the chat seems to be working fine.
<img width="1280" alt="Screenshot 2024-01-15 at 3 57 32 PM" src="https://github.com/Expensify/App/assets/35863227/b15ab5cd-a5c3-4339-b10c-17f4e6cf9090">

@aimane-chnaif Any thoughts?

**Comment by techievivek** [comment]
I think that should be fine since the report action handles it the same way for accountID with no personal details stored locally. But we can wait for @jasperhuangg to take a look into the bug and share his thoughts.

**Comment by jasperhuangg** [comment]
It looks like this is because of accountID-based mentions and that we are not handling them in the front-end for last message text.

This is definitely a back-end fix, and I believe this is a duplicate of https://github.com/Expensify/Expensify/issues/360572, which I plan on tackling today.

**Comment by sakluger** [comment]
@jasperhuangg are you sure this is a dupe of https://github.com/Expensify/Expensify/issues/360572? If so we should close this one in favor of the other issue.

**Comment by jasperhuangg** [comment]
Actually I'm not entirely sure if it's the same bug, sorry for the confusion.

The front-end should have all the necessary information in Onyx personal details once this is merged https://github.com/Expensify/Auth/pull/9689 @techievivek. So that proposal should work, but we'll need to hold it on this issue: https://github.com/Expensify/App/issues/34175

**Comment by techievivek** [comment]
Not overdue, holding working on the GH until https://github.com/Expensify/App/issues/34175 is done.

**Comment by sakluger** [comment]
I'm not sure - in https://github.com/Expensify/App/issues/34175, @jasperhuangg said that we should not implement a front-end fix because it would break 1:1:1, but that may be a different front-end fix.

@jasperhuangg what would you recommend as a next step for this issue?

**Comment by jasperhuangg** [comment]
The specific front end fix proposed by the contributor would break 1:1:1.

But I think we can still fix the front end to handle this correctly. We don't need to request the personal details here, we can just use the number of accountIDs in targetAccountIDs to say something like "invited X users".

**Comment by sakluger** [comment]
Got it - thanks @jasperhuangg for clarifying.

I reapplied the external label to make sure automation is working correctly and to hopefully attract some proposals. I'm also going to add this to vip-vsb since it seems most closely related to that project.

**Comment by sakluger** [comment]
Hey, @dukenv0307, it looks like you were offline for a few days for a bank holiday. Could you please respond to [this question](https://github.com/Expensify/App/issues/32978#issuecomment-1934951105) when you have a chance?

**Comment by sakluger** [comment]
Bump @dukenv0307.

**Comment by sakluger** [comment]
Hey @aimane-chnaif, could you please take a look at this one today?

**Comment by jasperhuangg** [comment]
I have an internal PR that should fix this already. This should be addressed in the back-end when the last message text should be generated.

**Comment by jasperhuangg** [comment]
We won't need a C+ for review here

**Comment by jasperhuangg** [comment]
The linked PR is on prod, just tested and seems to be working, closing

**Comment by jasperhuangg** [comment]
@aimane-chnaif can you tell me the reproduction steps used? Are they the same as in the OP?

**Comment by jasperhuangg** [comment]
Hmm, we shouldn't even be sending invite to room change logs in public rooms anymore

**Comment by jasperhuangg** [comment]
@aimane-chnaif I can't reproduce this on dev, are you reproducing this in staging?

**Comment by jasperhuangg** [comment]
Cool, it seems like we also need front-end fixes to address this, opening it back up for proposals

**Comment by sakluger** [comment]
Adding back the `Help Wanted` label, I think it was removed accidentally.

**Comment by DylanDylann** [comment]
would love to take over

**Comment by sakluger** [comment]
Thanks @DylanDylann! I've assigned you as C+.

@jasperhuangg please check the proposal when you have a chance.

**Comment by quinthar** [comment]
@dukenv0307 this was sent 3 days ago; can you give an ETA for completion?

**Comment by quinthar** [comment]
@dukenv0307 can you give an updated ETA?

**Comment by DylanDylann** [comment]
@dukenv0307 If there is any problem, please raise it here, we can discuss to speed up the PR

**Comment by quinthar** [comment]
@jasperhuangg will be back tomorow!

**Comment by jasperhuangg** [comment]
Still waiting for a review before moving forward with this

**Comment by jasperhuangg** [comment]
The Auth PR is on production and the front end is ready to be worked on for this!

**Comment by jasperhuangg** [comment]
@dukenv0307 let me know if you can work on this, or if you need anything else.

**Comment by jasperhuangg** [comment]
@dukenv0307 ah my mistake, I'll have some changes up for you shortly. There were some changes that we had that should have been applied to OpenApp but don't seem to be there.

**Comment by jasperhuangg** [comment]
@dukenv0307 I submitted a PR to update the response, I'll let you know when it's on staging so you can start using the change.

You'll need to sign in/sign out of your account in order for the change to take affect due to how our reliable updates work.

**Comment by jasperhuangg** [comment]
@dukenv0307 PR on staging

**Comment by sakluger** [comment]
I paid out both contracts. @DylanDylann please complete the BZ checklist.

**Comment by DylanDylann** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR: **NA**
 [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **NA**
 [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 [@DylanDylann] Determine if we should create a regression test for this bug. **Yes**
 [@DylanDylann] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


Regression Test Proposal
1. Create a workspace room.
2. Invite users.
3. Send a message to another chat (if on web).
4. Log out and log in.
5. Refresh the page.
6. Verify that: The last message text correctly shows "invited x users"


Do we agree 👍 or 👎

**Comment by sakluger** [comment]
Thanks! I made a few small tweaks (I suggested inviting exactly 2 users) to make the instructions more specific.
